File: pset1/cash/cash.c
// Let's save some coins!

#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)
{
    float g; //Amount owed
    int count = 0;
    do
    {
        // Prompt user for change owed
        g = get_float("Change owed: ");
    }
    while (g < 0);
    int t = roundf (g * 100);
    // How many quarters can be used and how much change left
    while (t >= 25)
    {
        count++; // The number of quarters used
        t -= 25;
    }
    // How many dimes can be used and how much change left
    while (t >= 10)
    {
        count++; // The number of dimes used
        t -= 10;
    }
    // How many nickels can be used and how much change left
    while (t >= 5)
    {
        count++; // The number of nickels used
        t -= 5;
    }
    // How many pennies can be used and how much change left
    while (t >= 1)
    {
        count++; // The number of pennies used
        t -= 1;
    }
    printf("We have used %i coin(s)!\n", count);
}

File: pset6/cash/cash.py
from cs50 import get_int
from cs50 import get_float

# Prompt user for change owed
while True:
    change = get_float("Change owed: ")
    if change > 0:
        break

t = round(change * 100)
# Initialize coin count
i = 0
# How many quarters can be used and how much change left
if t >= 25:
    i = t // 25
    t %= 25
# How many dimes can be used and how much change left
if t >= 10:
    i += t // 10
    t %= 10
# How many nickels can be used and how much change left
if t >= 5:
    i += t // 5
    t %= 5
# How many pennies can be used and how much change left
if t >= 1:
    i += t // 1
    t %= 1
print(f"We have used {i} coins.")
